src/theory/strings/theory_strings_rewriter.cpp

```diff
@@ -3436,21 +3455,22 @@ Node TheoryStringsRewriter::rewriteStringCode(Node n)
 
 Node TheoryStringsRewriter::splitConstant( Node a, Node b, int& index, bool isRev ) {
   Assert(a.isConst() && b.isConst());
-  index = a.getConst<String>().size() <= b.getConst<String>().size() ? 1 : 0;
-  unsigned len_short = index==1 ? a.getConst<String>().size() : b.getConst<String>().size();
+  size_t lenA = Word::getLength(a);
+  size_t lenB = Word::getLength(b);
+  index = lenA <= lenB ? 1 : 0;
+  unsigned len_short = index == 1 ? lenA : lenB;
```

Comment:
```suggestion
  size_t len_short = index == 1 ? lenA : lenB;
```

Might as well make that change.